presto-tests/src/test/java/com/facebook/presto/tests/TestTpchDistributedStats.java

```diff
@@ -26,9 +29,110 @@ public TestTpchDistributedStats()
     {}
 
     @Test
-    void testEstimateForSimpleQuery()
+    void testTableScanStats()
     {
-        assertEquals(compareMetric("SELECT * FROM NATION").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM nation").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM part").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM partsupp").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM region").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM supplier").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM lineitem").result(), MATCH);
+        assertEquals(compareMetric("SELECT * FROM lineitem").result(), MATCH);
     }
 
+    @Test
+    void testFilter()
+    {
+        String query = "" +
+                "SELECT * " +
+                "FROM lineitem " +
+                "WHERE l_shipdate <= DATE '1998-12-01' - INTERVAL '90' DAY";
+        MetricComparison metricComparison = compareMetric(query);
+        assertEquals(metricComparison.result(), DIFFER);
+        assertEquals(metricComparison.getEstimate().get(), 30000, 1);
```

Comment:
This is brittle, as the asserted range is not in any way related with the tolerance used for comparison in the `result` method.

Please introduce `MetricComparison#evaluate() : MetricComparison.Result` taking a function or some other form of a strategy object, and remove the `getEstimate` method. This way we can customize what is a MATCH and what is not in each test independently.

The strategy should only be used to decide betweeen MATCH and DIFFER and so it should take two double arguments (not two optionals)